internal/receiver/discoveryreceiver/receiver.go

```diff
@@ -57,29 +66,62 @@ func newDiscoveryReceiver(
 		logger:       settings.TelemetrySettings.Logger,
 		settings:     settings,
 		logsConsumer: consumer,
+		pLogs:        make(chan plog.Logs),
+		sentinel:     make(chan struct{}, 1),
+		loopFinished: &sync.WaitGroup{},
 	}
 
-	return d, nil
+	return d
 }
 
 func (d *discoveryReceiver) Start(ctx context.Context, host component.Host) (err error) {
 	if d.observables, err = d.observablesFromHost(host); err != nil {
 		return fmt.Errorf("failed obtaining observables from host: %w", err)
 	}
-	return nil
+
+	d.endpointTracker = newEndpointTracker(d.observables, d.config, d.logger, d.pLogs)
+	d.endpointTracker.start()
+
+	loopStarted := &sync.WaitGroup{}
+	loopStarted.Add(1)
+	d.loopFinished.Add(1)
+	go d.consumerLoop(loopStarted)
+	// wait until we know consumer loop is running before starting receiver creator
+	// so as not to miss any resulting telemetry
+	loopStarted.Wait()
```

Comment:
I might suggest putting a debug message around here about starting and waiting.

Reply:
good call, updated.